packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx

```diff
@@ -27,9 +27,10 @@ export type CommentInputProps = {
   memberHandle?: string
   onComment?: () => void
   onCancel?: () => void
-  onChange?: (e: React.ChangeEvent<HTMLTextAreaElement>) => void
+  onChange?: (value: string) => void
   onFocus?: () => void
   value?: string
+  initialValue?: string
```

Comment:
We need `initialValue` because we need to know if the comment being edited has changed from its `initialValue` before being able to publish it again

Reply:
I still sustain that you don't need that. You also don't need this awkward `useEffect` below.  
Hear me out:  
1. Remove initialValue and this useEffect
2. When a user clicks edit, we're calling: 
```
          onEditClick={() => setIsEditingComment({ commentId: comment.id, value: true })}
```
We could add here `setCommentInputTextRecord` and set the `comment.text`(basically what you do in this `useEffect`)
3. Instead of passing initialValue we could pass `isEditChanged` or something, which will be condition `commentInputTextRecord[commentId] === comment.text`
4. We're handling this `isEditChanged` inside the CommentInput and show a dialog if is false.